vignettes/bambu.Rmd

```diff
@@ -232,29 +273,139 @@ and the gene expression estimates that include any newly discovered genes and
+
+```{r}
+se <- bambu(reads = bamFiles, annotations = HsChr22BambuAnnotation, genome = genomeSequence, trackReads = TRUE)
+metadata(se)$readToTranscriptMaps[[1]]
+```
+
+
+
+### Training a model on another species/dataset and applying it
+
+In situations where training is not or cannot be performed, and the default model is also not suitable for the sample (the sample is generated from a different technology, species, condition, etc), *bambu* provides the option to train a new model, if well annotated similar data is available. For example one might train a model on arabidopsis to apply to an unannotated plant sample.
+
+```{r, eval = FALSE}
+# first train the model using a related annotated dataset from .bam
+se = bambu(reads = sample1.bam, annotations = annotations, genome = fa.file, discovery = FALSE, quant = FALSE, opt.discovery = list(returnModel = TRUE)) # note that discovery and quant need to be set to FALSE, alternatively you can have them set to TRUE and retrieve the model from the rcFile as long as returnModel = TRUE ([see here](#Storing-and-using-preprocessed-files-rcFiles)).
+newDefaultModel = metadata(se[[1]])$model # [[1]] will select the model trained on the first sample
+
+# alternatively train the model using an rcFile
+rcFile <- readRDS(pathToRcFile)
+newDefaultModel = trainBambu(rcFile)
+
+# use the trained model on another sample
+# sample2.bam and fa.file2 represent the aligned reads and genome for the poorly annotated sample
+se <- bambu(reads = sample2.bam, annotations = NULL, genome = fa.file2, opt.discovery = list(defaultModels = newDefaultModel, fitReadClassModel = FALSE))
+
+#trainBambu Arguments
+
+rcFile <- NULL, min.readCount = 2, nrounds = 50, NDR.threshold = 0.1, verbose = TRUE
+```
+
+
+### Including single exons
+
+By default *bambu* does not report single exon transcripts because they are known to have a high frequency of false positives and do not have splice junctions that are used by *bambu* to distinguish read classes. Nevertheless *bambu* trains a separate model on single-exon transcripts, and these predictions can be accessed and included in the annotations.
+
+```{r}
+se <- bambu(reads = bamFiles, annotations = HsChr22BambuAnnotation, genome = genomeSequence, opt.discovery = list(min.txScore.singleExon = 0))
+```
+
+## Downstream analysis 
+### Identifying differentially expressed genes {#DESeq2}
 One of the most common tasks when analysing RNA-Seq data is the analysis of 
 differential gene expression across a condition of intertest. Here we use 
 *DESeq2* to find the differentially expressed genes between MCF7 and K562 cell
 lines. Similar to using results from *Salmon*, estimates from *bambu* will
 first be rounded.
+
+
 ```{r}
 library(DESeq2)
-dds <- DESeqDataSetFromMatrix(round(assays(seGene)$counts),
-                                    colData = colData(se),
+dds <- DESeqDataSetFromMatrix(round(assays(seGene.multiSample)$counts),
```

Comment:
I couldn't test this with seGene.multiSample but it did work with se.multiSample